File: Pythonleetcode/Leetcode118.py
class Solution(object):
    def generate(self, numRows):
        ans=[[1]]
        if numRows==1:
            return ans
        if numRows == 0:
            return []
        for i in range(1,numRows):
                row=[1]*(i+1)
                ans.append(row)
                for j in range(0, i):
                    if j!=0:
                        ans[i][j]=ans[i-1][j-1]+ans[i-1][j]


        return ans

print(Solution().generate(0))

File: CppPractice/CppLeetcode/Leetcode118.cpp
class Solution {
public:
    vector<vector<int>> generate(int numRows) {
        vector<vector<int>> ans={{1},{1,1}};
        vector<int> temp={1,1};
        if(numRows==1){return {{1}};}
        if(numRows==2){return {{1},{1,1}};}
        for(int i=2;i<numRows;i++)
        {
            vector<int> newTemp = temp;
            newTemp.push_back(1);
            for(int j=1;j<newTemp.size()-1;j++){newTemp[j]=temp[j]+temp[j-1];}
            ans.push_back(newTemp);
            temp = newTemp;
            // cout<<temp<<endl;
        }
        return ans;
        
    }
};